### PR TITLE
Switch to yarn, include linting fixes, setup github actions

### PR DESCRIPTION
**Pull request by subdavis**
* swap npm for yarn
* upgrade eslint dependencies
* move eslintrc into package.json
* add jsconfig.json, which is a special vs code file.  It looks like there's a bug in VS code that causes it to ignore this file under certain conditions. You're supposed to be able to put it anywhere, but unless it's in the project's root (the "working directory") it seems to be fully or partially ignored.  If you open the `client` subfolder as the root dir, this works. 

The project now passes when you run `yarn lint`

Partial fix for #36

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "serve": "vue-cli-service serve",
+    "serve": "rimraf -rf ./node_modules/.cache/vue-loader && vue-cli-service serve",
```
Comment (AlmightyYakob) [inline client/package.json:6]: Why is this needed?

File: client/package.json [hunk 2]
```diff
@@ -27,20 +27,48 @@
...
+          "allow": [
+            "warn",
+            "error"
+          ]
+        }
+      ]
+    },
+    "parserOptions": {
+      "parser": "babel-eslint"
+    }
```
Comment (AlmightyYakob) [inline client/package.json:72]: Is there a particular reason you prefer to put the eslint config in `package.json`? I personally find having an `.eslintrc.js` to be more flexible, and allow more clarity due to comment support.

Comment (subdavis) [reply client/package.json:72]: Cross project consistency.  VS Code also seems to ignore .eslintrc.js when it isn't in the project root, as having it here resolved the `@` alias resolution linting errors.

I've found a lot of VS Code bugs today, actually.

File: client/package.json (hunk 1 above)
Comment (subdavis) [reply client/package.json:6]: Documented in `client/README.md`

File: client/src/components/Upload.vue [hunk 3]
```diff
@@ -60,8 +61,8 @@ async function readFilesFromDrop(e) {
     return [
       firstEntry.name,
       Array.from(e.dataTransfer.items)
-        .filter((item) => item.webkitGetAsEntry().isFile)
-        .map((item) => item.getAsFile()),
+        .filter(({ webkitGetAsEntry }) => webkitGetAsEntry().isFile)
+        .map(({ getAsFile }) => getAsFile()),
```
Comment (BryonLewis) [inline client/src/components/Upload.vue:65]: I'm getting an 'Illegal Invocation' error with `webkitGetAsEntry` and `getAsFile`.  This occurs when you drop a list of files onto the drag/drop area for uploading.  I'm assuming when you destructure a function from the object it can lose it's calling context if the invocation uses a (this) reference?   I don't know exactly how either `webkitGetAsEntry` or `getAsFile` function under the hood.  I believe this was a fix to the `no-shadow` linting requirement for `item`

Comment (subdavis) [reply client/src/components/Upload.vue:65]: Excellent catch.
